api.md

```diff
@@ -0,0 +1,63 @@
+# Kubernetes External Secrets Storage Frontends
+
+Kubernetes External Secrets supports several "frontends" for storing
+secret data and presenting it to applications:
+
+* [Secret frontend](#secret-frontend)
+* [Volume frontend](#volume-frontend)
+
+## Secret frontend
+
+TODO.
+
+## Volume frontend
+
+The volume frontend writes secret data to volumes included in `Pod`
+specs. You configure which volumes the External Secret controller
+writes data to by adding the
+`externalsecrets.kubernetes-client.io/volumes` annotation to a `Pod`
+manifest:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  generateName: pod-example-
+  annotations:
+    externalsecrets.kubernetes-client.io/volumes: >
+      [
+        {"name": "db-secrets", "externalSecretName": "db-secrets"},
```

Comment:
Another way I was thinking about this was to mirror volume declarations.  E.g.,

```json
{"name": "db-secrets", "externalSecret": { "externalSecretName": "db-secrets" } }
```

https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod

I thought that my be too verbose.  On the other hand it parallels an existing Kubernetes pattern.

Reply:
Maybe 
```
{"name": "db-secrets", "externalSecret": { "name": "db-secrets" } }
```
?

Reply:
I like that even better than externalSecretName

Reply:
Yeah, it makes more sense.

Reply:
I also like "name" better than "externalSecretName", but it departs from the kubernetes convention (see the link I included above), so I'd encourage us to follow the convention of go with "externalSecretName".